7a_temp_coop_munge/src/data_parsers/parse_UniversityofMissouri_LimnoProfiles_2017_2020.R

```diff
@@ -10,95 +10,64 @@ parse_UniversityofMissouri_LimnoProfiles_2017_2020 <- function(inind, outind) {
   files_from_zip <- unzip(infile, exdir = unzip_dir)
   on.exit(unlink(unzip_dir, recursive = TRUE))
 
-  # Set grouping variables and define relevant column names for each data set -------------
-
   # Files fall into roughly 5 categories, as defined below.
   # A few notes on the files in `files_2017_hw`:
   # (1) `~/092_08-02-2017 HW.csv` is duplicated as `~/092_08_02_2017 HW.csv`.
   # (2) The file format for both files in (1) doesn't match any other file in the 2017 HW files
   # Issues (1) and (2) are resolved by removing them both from the `files_2017_hw`
   # and then adding one (`092_08_02_2017 HW.csv`) into it's own vector
-  files_2017 <- files_from_zip[grepl("_2017_.*cleaned", files_from_zip)]
-  files_2018_2020 <- files_from_zip[grepl('_2018.*cleaned|_2019|_2020',files_from_zip)]
-  files_2017_hw <- files_from_zip[grepl('_2017_', files_from_zip)] %>%
-    .[grepl('HW', .)] %>%
-    .[!grepl('092_08_02_2017|092_08-02-2017', .)]
-  files_2017_092_hw <-  files_from_zip[grepl('092_08_02_2017 HW.csv', files_from_zip)]
-  files_2018_hw <- files_from_zip[grepl('_2018.*HW', files_from_zip)]
+  files_tbl <- tibble(filepath = files_from_zip) %>%
+    mutate(func_name = case_when(
+      str_detect(filepath, '_2017_.*cleaned') ~
+        'read_files_2017',
+      str_detect(filepath, '_2018.*cleaned|_2019|_2020') ~
+        'read_files_2018_2020',
+      str_detect(filepath, '_2017_') & str_detect(filepath, 'HW') & !str_detect(filepath, '092_08_02_2017|092_08-02-2017') ~
+        'read_files_2017_hw',
+      str_detect(filepath, '092_08_02_2017 HW.csv') ~
+        'read_files_2017_092_hw',
+      str_detect(filepath, '_2018.*HW') ~
+        'read_files_2018_hw'
+      # if a file doesn't below to a group, this column will be NA:
+    )) %>% filter(!is.na(func_name))
```

Comment:
Very into this approach of making a table

Reply:
Cool. I think a table can also hold other useful things, such as arguments that go into the function that handles the file. But in this case, that didn't seem to simplify things because the number of args in the custom functions are going to differ.